sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/ScalaReflection.scala

```diff
@@ -806,25 +807,38 @@ object ScalaReflection extends ScalaReflection {
     encoderFor(typeTag[E].in(mirror).tpe).asInstanceOf[AgnosticEncoder[E]]
   }
 
+  /**
+   * Same as [[encoderFor]] but with extended support to return [[UnboundRowEncoder]] for [[Row]]
+   * type.
+   */
+  def encoderForWithRowEncoderSupport[E: TypeTag]: AgnosticEncoder[E] = {
+    encoderFor(typeTag[E].in(mirror).tpe, isRowEncoderSupported = true)
```

Comment:
Is `isRowEncoderSupported` ever read? It looks to me like it just passed around but never actually read.